test/unit/util_cast.js

```diff
@@ -0,0 +1,178 @@
+var test = require('tap').test;
+var cast = require('../../src/util/cast');
+
+test('toNumber', function (t) {
+    // Numeric
+    t.equal(cast.toNumber(0), 0);
+    t.equal(cast.toNumber(1), 1);
+    t.equal(cast.toNumber(3.14), 3.14);
+
+    // String
+    t.equal(cast.toNumber('0'), 0);
+    t.equal(cast.toNumber('1'), 1);
+    t.equal(cast.toNumber('3.14'), 3.14);
+    t.equal(cast.toNumber('0.1e10'), 1000000000);
+    t.equal(cast.toNumber('foobar'), 0);
+
+    // Boolean
+    t.equal(cast.toNumber(true), 1);
+    t.equal(cast.toNumber(false), 0);
+    t.equal(cast.toNumber('true'), 0);
+    t.equal(cast.toNumber('false'), 0);
+
+    // Undefined & object
+    t.equal(cast.toNumber(undefined), 0);
+    t.equal(cast.toNumber({}), 0);
+    t.end();
```

Comment:
May want to do `t.equal(cast.toNumber(NaN), 0);` since that's specifically dealt with in the function.